python/pyarrow/types.pxi

```diff
@@ -2154,6 +2154,14 @@ def duration(unit):
     return out
 
 
+def month_day_nano_interval():
+    """
+    Create instance of a interval representing the time between two calendar
+    instances represented as a triple of months, days and nanoseconds.
+    """
+    return primitive_type(_Type_INTERVAL_MONTH_DAY_NANO)
```

Comment:
Nit: I would avoid "the time between two calendar instances" because a duration/Joda-interval is traditionally defined as "the time between two instants".  I haven't seen intervals / periods defined in this way before (possibly because calendars could change).

Also, instead of `a triple of months` (which might make a python user think 3-tuple) could we say "represented by values for months, days, and nanoseconds"?  Or, if we want to be real precise, "represented by a signed 32 bit integer of months, a signed 32 bit integer of days, and a signed 64 bit integer of nanoseconds".

Reply:
copied language from type.h